resources/models.py

```diff
@@ -252,6 +263,49 @@ def __str__(self):
         return self.safe_translation_getter("name", super().__str__())
 
 
+class AbstractAreaMap(UUIDModel):
+    map_file = models.FileField(
+        upload_to=get_map_resource_media_folder,
+        storage=OverwriteStorage(),
```

Comment:
I think using the existing `OverwriteStorage` here would mean that when you upload a harbor map or even image_file for the harbor, the entire `/harbors/{harbor_id}/` folder would get removed and created again. I think we do not need `OverwriteStorage` anymore. Folder should remain 4eva now. I see you made some signal handler, so those should suffice.